From e02394d30b6cc2e09b3eba4c4da3c0fe80ad9d13 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthana.kalyanasundaram@broadcom.com>
Date: Tue, 12 Nov 2024 06:16:56 +0000
Subject: [PATCH] canister: Change spinlock_t lock to void *reserved

change tyope of variable lock in structure aead_geniv_ctx from spinlock_t
to void * similar to fips_canister to maintain the compatibility. Rename
the lock variable to reserved for canister usage (for the remaining kernel).
---
 include/crypto/internal/geniv.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/crypto/internal/geniv.h b/include/crypto/internal/geniv.h
index 7fd7126f5..d54724057 100644
--- a/include/crypto/internal/geniv.h
+++ b/include/crypto/internal/geniv.h
@@ -13,7 +13,7 @@
 #include <linux/types.h>
 
 struct aead_geniv_ctx {
-	spinlock_t lock;
+	void *reserved;
 	struct crypto_aead *child;
 	struct crypto_sync_skcipher *sknull;
 	u8 salt[] __attribute__ ((aligned(__alignof__(u32))));
-- 
2.39.4

